### ISSUE TITLE
Make Doctrine column string lengths dynamic

### ISSUE THREAD

**Issue by ro0NL** [msgphp/msgphp#159]
Something like "if collation is `utf8mb4_unicode_ci` then string length is 191, else 255"

**Comment by kniziol** [comment]
It's related to `utf8mb4_unicode_ci` and `MySQL 5.6`. Full error while running `bin/console doctrine:schema:create --env=dev` command:

```console
  Schema-Tool failed with Error 'An exception occurred while executing 'CREAT
  E TABLE user (id INT AUTO_INCREMENT NOT NULL COMMENT '(DC2Type:msgphp_user_
  id)', credential_email VARCHAR(255) NOT NULL, credential_password VARCHAR(2
  55) NOT NULL, password_reset_token VARCHAR(255) DEFAULT NULL, password_requ
  ested_at DATETIME DEFAULT NULL, UNIQUE INDEX UNIQ_8D93D649A5D24B55 (credent
  ial_email), UNIQUE INDEX UNIQ_8D93D6496B7BA4B6 (password_reset_token), PRIM
  ARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci ENGIN
  E = InnoDB':
```

How to increase field size right now?

**Comment by ro0NL** [comment]
@kniziol seems like https://github.com/FriendsOfSymfony/FOSUserBundle/issues/1919#issuecomment-217759629 is good workaround.

Im not sure if anything needs to be done... it works as of MySQL 5.7

**Comment by ro0NL** [comment]
Favoring application override / latest version for now.

**Comment by ro0NL** [comment]
@kniziol fixed in 0.8 by default, or to be customized using e.g.

```yaml
parameters:
  msgphp.doctrine.mapping_config
    key_max_length: 100
```

per case xml overrides for the core mappings can be put in `config/packages/msgphp/doctrine`
